nautobot/extras/choices.py

```diff
@@ -251,6 +251,54 @@ class RelationshipTypeChoices(ChoiceSet):
     )
 
 
+#
+# Secrets
+#
+
+
+class SecretsGroupAccessTypeChoices(ChoiceSet):
```

Comment:
The hope was that we could keep the design simpler by not introducing two additional data models here. Additionally, they kind of need to be constants because we need to be able to look them up in some fashion - if the user can dynamically define which AccessType means "http(s)" (or even not define that type at all), how can a GitRepository look up the HTTP(S) secret that it needs?

Reply:
That's fair, but it is important to note we already have this problem with status, related to the imbedded logic for `slacc` and `container`. Not saying we should perpetuate that problem, but something to think about.